### ISSUE TITLE
CreateSVGDocument takes a C string

### ISSUE THREAD

**Issue by bungeman** [adobe/svg-native-viewer#164]
CreateSVGDocument should take a pointer and a length for the SVG data, as well as document that the encoding of the data must be UTF-8. While the XML specification forbids the use of U+0000 so it could in theory be used as a stop, real UTF-8 is not terminated by U+0000 (U+0000 is a perfectly legal code point to have in the middle of a UTF-8 string). As a result, users may have the SVG document in UTF-8 in hand but without any trailing '\0', forcing a copy just to add this extra byte.

This is additionally interesting because the sane xml parsers ([Expat](https://github.com/adobe/svg-native-viewer/blob/f5c2ab195c6f3eaa76c2037817fd0d21b505bbc1/svgnative/src/xml/ExpatXMLParser.cpp#L101), [LibXML2](https://github.com/GNOME/libxml2/blob/0137d9879b08a243f8125b764d07f8f86dd3998d/parser.c#L14536)) are taking this '\0' terminated string an immediately calling strlen on it. Note that LibXML2's xmlReadMemory should be used instead of xmlReadDoc. [RapidXML](https://code.woboq.org/boost/boost/boost/property_tree/detail/rapidxml.hpp.html) is not sane and actually does require a '\0' terminated string, and even makes a note in its comments that it is unsuitable for parsing straight mmap'ed files. If RapidXML is to continue to be used, then there will need two ways of providing the SVG data, one as a '\0' terminated string and one as a pointer and length (and the RapidXML port will just need to make a copy) or there will need to be some way to indicate to the user which is better.

**Comment by dirkschulze** [comment]
Good point. We should add the length argument with additional information how to use the API in case of RapidJSON. Not sure why we didn't use `xmlReadMemory` earlier. Need to look into it.

**Comment by HinTak** [comment]
I found myself needing this too. The OT-SVG data coming out of freetype is not (necessarily) null-terminated, but does have an accompanying length description.
